### DISCUSSION TITLE
ecef to wgs84

### DISCUSSION THREAD

**Discussion by luohuiiqng** [pelicanmapping/osgearth#2488] [Support Q&A] [answered]
Hello, I am trying to convert the ecef coordinates to wgs84. The code is as follows:
```
void ecef2Wgs84LAT(osg::Vec3d src, osg::Vec3d& outlat) {
		osg::ref_ptr<osgEarth::SpatialReference> ecefSRS = osgEarth::SpatialReference::create("epsg:4978");
		osg::ref_ptr<osgEarth::SpatialReference> wgs84SRS = osgEarth::SpatialReference::create("wgs84");
		osgEarth::GeoPoint point(osgEarth::SpatialReference::create("epsg:4978"), src);
		osg::Vec3d world;
		point.toWorld(world);
		wgs84SRS->transformFromWorld(world, outlat);
	}
```
When I pass in the coordinates of ecef as (-1975.254792, -3609.488278, -6252.416188), The result I get is (-118.6892,-56.8018,1121637.77781), which is different from the result I get from some online coordinate conversion tool, which gets (-56.80184(latitude),241.31078,1121637.8). What improvements do I need to make? Thank you!

**Comment by gwaldron** [comment, marked as answer]
The canonical way to convert between WG84 and ECEF is like so:
```c++
auto wgs84 = SpatialReference::get("wgs84");
auto ecef = wgs84->getGeocentricSRS();

GeoPoint ecef_point(ecef_srs, src);
GeoPoint wgs84_point = ecef_point.transform(wgs84);
```

Also, your test ECEF coordinates look out of range to me.

**Comment by luohuiiqng** [comment]
Thanks you!
